src/plugins/usage_collection/server/collector/collector_set.ts

```diff
@@ -24,50 +24,63 @@ import {
   ElasticsearchClient,
   ISavedObjectsRepository,
   SavedObjectsClientContract,
+  KibanaRequest,
 } from 'kibana/server';
 import { Collector, CollectorOptions } from './collector';
 import { UsageCollector, UsageCollectorOptions } from './usage_collector';
 
+type AnyCollector = Collector<any, any>;
+type AnyUsageCollector = UsageCollector<any, any>;
+
 interface CollectorSetConfig {
   logger: Logger;
   maximumWaitTimeForAllCollectorsInS?: number;
-  collectors?: Array<Collector<any, any>>;
+  collectors?: AnyCollector[];
 }
 
 export class CollectorSet {
   private _waitingForAllCollectorsTimestamp?: number;
   private readonly logger: Logger;
   private readonly maximumWaitTimeForAllCollectorsInS: number;
-  private readonly collectors: Map<string, Collector<any, any>>;
+  private readonly collectors: Map<string, AnyCollector>;
   constructor({ logger, maximumWaitTimeForAllCollectorsInS, collectors = [] }: CollectorSetConfig) {
     this.logger = logger;
     this.collectors = new Map(collectors.map((collector) => [collector.type, collector]));
     this.maximumWaitTimeForAllCollectorsInS = maximumWaitTimeForAllCollectorsInS || 60;
   }
 
   public makeStatsCollector = <
-    T,
-    U,
-    O extends CollectorOptions<T, U> = CollectorOptions<T, U> // Used to allow extra properties (the Collector constructor extends the class with the additional options provided)
+    TFetchReturn,
+    TFormatForBulkUpload,
+    WithKibanaRequest extends boolean
   >(
-    options: O
+    options: CollectorOptions<TFetchReturn, TFormatForBulkUpload, WithKibanaRequest>
   ) => {
-    return new Collector(this.logger, options);
+    return new Collector<TFetchReturn, TFormatForBulkUpload>(this.logger, options);
   };
   public makeUsageCollector = <
-    T,
-    U = T,
-    O extends UsageCollectorOptions<T, U> = UsageCollectorOptions<T, U>
+    TFetchReturn,
+    TFormatForBulkUpload = { usage: { [key: string]: TFetchReturn } },
+    // TODO: Right now, users will need to explicitly claim `true` for TS to allow `kibanaRequest` usage.
+    //  If we improve `telemetry-check-tools` so plugins do not need to specify TFetchReturn,
+    //  we'll be able to remove the type defaults and TS will successfully infer the config value as provided in JS.
+    WithKibanaRequest extends boolean = false
   >(
-    options: O
+    options: UsageCollectorOptions<TFetchReturn, TFormatForBulkUpload, WithKibanaRequest>
   ) => {
-    return new UsageCollector(this.logger, options);
+    return new UsageCollector<TFetchReturn, TFormatForBulkUpload>(this.logger, options);
   };
 
   /*
    * @param collector {Collector} collector object
    */
-  public registerCollector = <T, U>(collector: Collector<T, U>) => {
+  public registerCollector = <
+    TFetchReturn,
+    TFormatForBulkUpload,
+    WithKibanaRequest extends boolean
```

Comment:
`WithKibanaRequest` isn't used in `registerCollector`.

Reply:
Good catch! Removed!